### PR TITLE
Refactor Template detail table to use Inventory table

### PR DESCRIPTION
**Pull request by leSamo**
https://issues.redhat.com/browse/SPM-1975

There should be almost no change in functionality, aside from added Tags column and sortable OS column in Template detail table.

### CODE REVIEW

**Review by mkholjuraev — CHANGES_REQUESTED** [review]
Codewise looks good to me. A few comments I have faced an issue while searching for a system in the template review systems step that needs to be fixed.

image

File: src/Utilities/api.js [hunk 1]
```diff
@@ -202,8 +202,11 @@ export const fetchPatchSet = id => {
     return createApiCall(`/baselines/${id}`, 'v3', 'get');
 };
 
-export const fetchPatchSetSystems = (id, params) => {
-    return createApiCall(`/baselines/${id}/systems`, 'v3', 'get', params);
+export const fetchPatchSetSystems = (params) => {
+    const id = params.id;
```
[the remark this answers is not in the document]
Comment (leSamo) [reply src/Utilities/api.js:206]: I wish I didn't need to do this, but `getEntities` function is setup this way along with all other endpoints using InventoryTable, so I'll either refactor this single function or refactor 4 or so other endpoints along with `getEntities` function 😢

File: src/SmartComponents/PatchSetDetail/PatchSetDetail.js [hunk 2]
```diff
@@ -56,49 +64,39 @@ const PatchSetDetail = () => {
...
+    const loaded = useSelector(
+        ({ entities }) => entities?.loaded
     );
 
     const queryParams = useSelector(
-        ({ PatchSetDetailSystemsStore }) => PatchSetDetailSystemsStore.queryParams
+        ({ PatchSetDetailSystemsStore }) => PatchSetDetailSystemsStore?.queryParams || {}
     );
 
+    //console.log("aaaaaaaaaaaaaaaaa", a, totalItems);
```
[the remark this answers is not in the document]
Comment (leSamo) [reply src/SmartComponents/PatchSetDetail/PatchSetDetail.js:87]: Whoops :sweat_smile:

File: src/SmartComponents/PatchSetDetail/PatchSetDetail.js [hunk 3]
```diff
@@ -107,7 +105,10 @@ const PatchSetDetail = () => {
     };
 
     const refreshTable = () => {
-        dispatch(fetchPatchSetDetailSystemsAction(patchSetId, { ...queryParams, page: 1, offset: 0 }));
+        // timestamp is used to force inventory to refresh
+        // if it wasn't there inventory might ignore request to refresh because parameters are the same
+        inventory?.current?.onRefreshData({ timestamp: Date.now() });
```
[the remark this answers is not in the document]
Comment (leSamo) [reply src/SmartComponents/PatchSetDetail/PatchSetDetail.js:110]: Good question! No it doesn't, it's filtered out.

File: src/SmartComponents/PatchSetDetail/PatchSetDetail.js [hunk 4]
```diff
@@ -141,16 +141,14 @@ const PatchSetDetail = () => {
         } else {
             history.push(encodeURLParams(queryParams));
 
-            dispatch(fetchPatchSetDetailSystemsAction(patchSetId, queryParams));
+            //dispatch(fetchPatchSetDetailSystemsAction(patchSetId, queryParams));
```
[the remark this answers is not in the document]
Comment (leSamo) [reply src/SmartComponents/PatchSetDetail/PatchSetDetail.js]: Good catch, I removed that line.

**Comment by leSamo** [comment]
@mkholjuraev thanks for review! I addressed all your comments.

**Review by AsToNlele — APPROVED** [review]
With the fix works great, good job! Only a small nitpick "Upto" should propably be "Up to"
image

**Pull request merged**